### ISSUE TITLE
Include arm64 image in tests

### ISSUE THREAD

**Issue by pmalek** [Kong/gateway-operator#195]
As of now, we're sort of relying on `amd64` for tests, dismissing the notion of differences between `amd64` and `arm64`.

In rare, corner cases those might exist and lead to runtime errors. We should include using `arm64` in e2e and image tests.

### Acceptance criteria

- [ ] `arm64` is added to e2e tests as equal with `amd64`

**Comment by pmalek** [comment]
But as I look at it know, it should really be a problem to build an arm64 image using docker/buildx/qemu.

**Comment by shaneutt** [comment]
It's effectively a nice-to-have since we don't have any _specific_ requirements driving it, however I think it would be very nice to get this done before [OSS Release v1](https://github.com/Kong/gateway-operator/milestone/28) as this will for a small upfront cost make it even easier to adopt Kong.

**Comment by mflendrich** [comment]
Blocked by availability of a convenient way to test on ARM using our existing CI system (or our desire to put some infra together by hand)

**Comment by shaneutt** [comment]
They are on their way: https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/
